app/src/androidTest/java/org/mozilla/fenix/ui/robots/DownloadRobot.kt

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("TooManyFunctions")
+
+package org.mozilla.fenix.ui.robots
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.RootMatchers.isDialog
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.Until
+import org.hamcrest.CoreMatchers
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.click
+import org.mozilla.fenix.helpers.ext.waitNotNull
+
+/**
+ * Implementation of Robot Pattern for download UI handling.
+ */
+class DownloadRobot {
+    fun verifyDownloadPrompt() = assertDownloadPrompt()
```

Comment:
@AaronMT this hasn't been resolved yet.  You mentioned you had some local fixes that you resolved as you worked on them and that you'd thought you had pushed and squashed.  It's probably a better practice to resolve things once they're up on github otherwise you might resolve comments that were never pushed.  Once you've addressed these, I'll come back and take another look.

Reply:
I've updated the PR to adhere to style guidelines.